packages/openapi3/test/union-schema.test.ts

```diff
@@ -619,8 +619,7 @@ worksFor(["3.0.0"], ({ diagnoseOpenApiFor, oapiForModel, openApiFor }) => {
         x: {
           anyOf: [
             {
-              type: "object",
-              allOf: [{ $ref: "#/components/schemas/MyStr" }],
+              type: "string",
```

Comment:
@chrisradek what do you think here? I am not sure we want to loose the ref, I think this should produce

```
type: string
nullable: true
allOf: [{ $ref: "#/components/schemas/MyStr" }]

```

Reply:
I think technically, unless `MyStr` also has `nullable: true` on it, this wouldn't work according to the spec. `nullable` only applies to its own schema, not subschemas.

This is one of those things that wasn't stated in the original 3.0.0 spec version, but was clarified later (at least in 3.0.4). I think prior to the clarification some tooling did actually support what you're proposing too.

Could try some tools/validators to see how they actually treat this, otherwise might need to copy the ref and add nullable to it :-/